app/Platform/Actions/UpdateGameSetFromGameAlternativesModification.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\Game;
+use App\Models\GameSet;
+use App\Models\System;
+use App\Platform\Enums\GameSetType;
+
+class UpdateGameSetFromGameAlternativesModification
+{
+    public function execute(int $parentGameId, int $childGameId, bool $isAttaching = true): void
+    {
+        $parentGame = Game::find($parentGameId);
+        $childGame = Game::find($childGameId);
+
+        // Determine if a swap is needed to ensure the hub is always the parent or consistent ordering.
+        if (
+            ($parentGame->ConsoleID !== System::Hubs && $childGame->ConsoleID === System::Hubs)
+            || ($parentGameId > $childGameId)
+        ) {
+            // Swap to ensure the hub is the parent, or ensure consistent ordering.
+            $temp = $parentGame;
+            $parentGame = $childGame;
+            $childGame = $temp;
+        }
+
+        $parentGameSet = GameSet::firstWhere('game_id', $parentGameId);
+        if (!$parentGameSet) {
+            $isParentGameHub = $parentGame->ConsoleID === System::Hubs;
+
+            $parentGameSet = GameSet::updateOrCreate(
+                ['game_id' => $parentGame->id],
+                [
+                    'title' => $isParentGameHub ? $parentGame->title : 'Similar Games',
+                    'type' => $isParentGameHub ? GameSetType::HUB : GameSetType::SIMILAR_GAMES,
+                    'image_asset_path' => $isParentGameHub ? $parentGame->ImageIcon : null,
+                ]
+            );
```

Comment:
```
Call to a member function games() on null
```
Trying to link game 12346 from game 12345.

Reply:
Resolved in latest.